### ISSUE TITLE
Permanent "cannot deliver" Icon

### ISSUE THREAD

**Issue by Murk78** [MaFi-Games/Captain-of-Industry-issues#340]
The Icon above some trucks that indicates that it can't dump it's load won't go away.

As soon as the truck would be able to dump it's load, it should dissappear.

If trucks are waiting at their mining control tower because they can't dump their load and you deconstruct that tower, the trucks will constantly have the icon that they can't deliver their load.
They can continue their work normally but they will have this icon and the message in their information window.

Game Version EA 0.4.4

![Bug](https://user-images.githubusercontent.com/107458601/173506917-3ebd300a-89ab-4910-8030-7eb7eced1103.jpg)

The trucks in the screenshot are working as they should even if the icon is still there.

**Comment by MaFi-Games** [comment]
I think I have found a repro case

**Comment by Murk78** [comment]
Now I can't even reproduce it like this morning...
All affected trucks belonged to my coalmine tower. I build an new one withe the same settings, deleted the old one and assigned the trucks to the new one. 10minutes later I saw the iconproblem. I tried it at another mine and was pretty sure that I had reproduced it. Maybe it was a coincidence
[Savegame.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/8900823/Savegame.zip)
